code/modules/events/sentience.dm

```diff
@@ -0,0 +1,42 @@
+/datum/round_event_control/sentience
+	name = "Sentience"
+	typepath = /datum/round_event/ghost_role/sentience
+	weight = 5
+
+/datum/round_event/ghost_role/sentience
+	minimum_required = 1
+	role_name = "random animal"
+
+/datum/round_event/ghost_role/sentience/spawn_role()
+	var/list/mob/dead/observer/candidates
+	candidates = get_candidates(ROLE_ALIEN, null, ROLE_ALIEN)
+
+	// find our chosen mob to breathe life into
+	// Mobs have to be simple animals, mindless and on station
+	var/list/potential = list()
+	for(var/mob/living/simple_animal/L in living_mob_list)
+		var/turf/T = get_turf(L)
+		if(T.z != ZLEVEL_STATION)
+			continue
+		if(!(L in player_list) && !L.mind)
+			potential += L
+
+	var/mob/living/simple_animal/SA = pick(potential)
+	var/mob/dead/observer/SG = pick(candidates)
+
+	if(!SA || !SG)
+		return FALSE
+
+	SA.key = SG.key
+	SA.languages_spoken |= HUMAN
+	SA.languages_understood |= HUMAN
+	SA.sentience_act()
+
+	SA.maxHealth = min(SA.maxHealth, 200)
```

Comment:
You probably meant max here, all this does is reduce health of megafauna if it gains sentience.